test/unittests/commands/excel2json/test_lists_compliance_checks.py

```diff
@@ -0,0 +1,122 @@
+        expected = {
+            "missing columns for list name": "There is not column with the expected format for the list names: "
+            "'[lang]_list'"
+        }
+        assert _check_min_num_col_present(test_cols) == expected
+
+    def test_missing_columns_node(self) -> None:
+        test_cols = pd.Index(["ID (optional)", "en_list"])
+        expected = {
+            "missing columns for nodes": "There is not column with the expected format for the list nodes: "
+            "'[lang]_[column_number]'"
+        }
+        assert _check_min_num_col_present(test_cols) == expected
+
+
+class TestCheckMinimumRows:
+    def test_good(self) -> None:
+        test_df = pd.DataFrame({"one": [1, 2, 3], "two": [4, 5, 6]})
+        assert not _check_minimum_rows(test_df)
+
+    def test_missing_rows(self) -> None:
+        test_df = pd.DataFrame({"one": [1]})
+        expected = {
+            "minimum rows": "The excel must contain at least two rows, "
+            "one for the list name and one row for a minimum of one node."
+        }
+        assert _check_minimum_rows(test_df) == expected
+
+
+class TestCheckWarnUnusualColumns:
+    def test_good(self) -> None:
+        test_cols = pd.Index(["ID (optional)", "en_list", "en_2", "de_2"])
+        with warnings.catch_warnings(record=True) as caught_warnings:
+            _check_warn_unusual_columns(test_cols)
+        assert len(caught_warnings) == 0
+
+    def test_additional_columns(self) -> None:
+        test_cols = pd.Index(["ID (optional)", "en_list", "en_2", "de_2", "additional_1", "additional_2"])
+        expected = regex.escape(
+            "The following columns do not conform to the expected format "
+            "and will not be included in the output: additional_1, additional_2"
+        )
+        with pytest.warns(DspToolsUserWarning, match=expected):
+            _check_warn_unusual_columns(test_cols)
+
+
+class TestCheckAllTranslationsPresent:
+    def test_good(self) -> None:
+        test_cols = pd.Index(["ID (optional)", "en_list", "de_list", "de_1", "en_1", "de_2", "en_2"])
+        assert not _check_all_expected_translations_present(test_cols)
+
+    def test_missing_translations_node_columns(self) -> None:
+        test_cols = pd.Index(["ID (optional)", "en_list", "de_list", "de_1", "en_1", "de_2"])
+        expected = {
+            "missing translations": "All the nodes must be translated into the same languages, "
+            "the translations for the following column(s): "
+            "en_2"
+        }
+        assert _check_all_expected_translations_present(test_cols) == expected
+
+    def test_missing_translations_list_columns(self) -> None:
+        test_cols = pd.Index(["ID (optional)", "en_list", "de_1", "en_1", "de_2", "en_2"])
+        expected = {
+            "missing translations": "All the nodes must be translated into the same languages, "
+            "the translations for the following column(s): "
+            "de_list"
```

Comment:
as already commented upon in the production code, I don't think that this message is understandable for the user.